plugin.video.npogemist/addon.py

```diff
@@ -0,0 +1,376 @@
+        info["dateadded"] = dateconversions.time2kodi_datetime(item["aired"])
+    list_item.setInfo("video", info)
+
+    if "duration" in item:
+        stream = {"duration": item["duration"]}
+        list_item.addStreamInfo("video", stream)
+
+    if item.get("is_video", False):
+        list_item.setProperty("IsPlayable", "true")
+    return list_item
+
+
+def display_broadcasts(broadcasts, add_time=False):
+    """Display a list of broadcasts."""
+    for broadcast in broadcasts:
+        directory.add(router.make_url(play, mid=broadcast["mid"]), make_list_item(broadcast, add_time=add_time),
+                      is_folder=False)
+    directory.display(DATE_DURATION_TITLE)
+
+
+def display_programs(programs, sortable=True):
+    """Display a list of programs."""
+    for program_item in programs:
+        directory.add(router.make_url(program, mid=program_item["mid"]), make_list_item(program_item))
+    sort_methods = []
+    if sortable:
+        sort_methods.append(xbmcplugin.SORT_METHOD_VIDEO_TITLE)
+    directory.display(sort_methods)
+
+
+def display_mixed(items):
+    """Display a list of episodes and programs, mixed."""
+    for item in items:
+        if item.get("is_video", False):
+            directory.add(router.make_url(play, mid=item["mid"]), make_list_item(item),
+                          is_folder=False)
+        else:
+            directory.add(router.make_url(program, mid=item["mid"]), make_list_item(item))
+    sort_methods = DATE_DURATION_TITLE
+    directory.display(sort_methods)
+
+
+@router.route("/play")
+def play(mid):
+    """Play a video with id mid."""
+    # store recent
+    # TODO If we don't have a serie, we want to store the episode, but we need to adapt recents for that.
+    # For now, skip if no series.
+    try:
+        serie_id = npo.get_serie_id_by_episode(mid)
+    except NPOContentError:
+        serie_id = None
+    if serie_id is not None:
+        if serie_id in profile_recent_programs:
+            profile_recent_programs.remove(serie_id)
+        profile_recent_programs.append(serie_id)
+        # store
+        profile.set_data("recent_programs", profile_recent_programs)
+    # now play
+    try:
+        play_item = xbmcgui.ListItem(path=(npo.get_play_url(mid)))
+        # Add subtitles
+        subs_url = npo.get_subtitles_url(mid)
+        play_item.setSubtitles([subs_url])
+        xbmcplugin.setResolvedUrl(handle, True, listitem=play_item)
+        # Wait for the player to play
+        player = playing_player()
+        # Enable/disable the subtitles
+        player.showSubtitles(setting_subtitles)
+    except NPOContentError:
+        xbmcgui.Dialog().notification(s(pypo.strings.npo_gemist),
+                                      s(pypo.strings.this_program_is_not_available), xbmcgui.NOTIFICATION_ERROR)
+
+
+def playing_player():
+    """Get the currently playing player, to add subtitles."""
+    player = xbmc.Player()
+    passed = 0
+    while not player.isPlaying():
```

Comment:
AFIK, this won't work correctly for Leia

Reply:
true, please drop all this logic and use only setSubtitles if the setting is true

Reply:
That wouldn't give the same functionality.
In the current implementation, the user can set the setting to false, but then still switch on the subtitles from within the player. If I only use setSubtitles if the setting is true, the user will have a greyed-out setting in the player; they would need to leave the player, change the setting and enter the player again.
Is there another way to change the player's default?